pypowerwall/tedapi/vitals_dictionary.py

```diff
@@ -0,0 +1,517 @@
+                                "cts": cts,
+                                "inverted": meter.get('inverted'),
+                                "connection": meter.get('connection'),
+                                "real_power_scale_factor": meter.get('real_power_scale_factor', 1)
+                            }
+        return meter_config
+
+    def get_neurio(self):
+        # Build meter Lookup if available
+        meter_config = self.get_meters()
+        neurio = {}
+        c = 1000
+        # Loop through each Neurio device serial number
+        for n in lookup(self.status, ['neurio', 'readings']) or {}:
+            # Loop through each CT on the Neurio device
+            sn = n.get('serial', str(c))
+            cts = {}
+            c = c + 1
+            for i, ct in enumerate(n['dataRead'] or {}):
+                # Only show if we have a meter configuration and cts[i] is true
+                cts_bool = lookup(meter_config, [sn, 'cts'])
+                if isinstance(cts_bool, list) and i < len(cts_bool):
+                    if not cts_bool[i]:
+                        # Skip this CT
+                        continue
+                factor = lookup(meter_config, [sn, 'real_power_scale_factor']) or 1
+                device = f"NEURIO_CT{i}_"
+                cts[device + "InstRealPower"] = lookup(ct, ['realPowerW']) * factor
+                cts[device + "InstReactivePower"] = lookup(ct, ['reactivePowerVAR'])
+                cts[device + "InstVoltage"] = lookup(ct, ['voltageV'])
+                cts[device + "InstCurrent"] = lookup(ct, ['currentA'])
+                location = lookup(meter_config, [sn, 'location'])
+                cts[device + "Location"] = location[i] if len(location) > i else None
+            meter_manufacturer = None
+            if lookup(meter_config, [sn, 'type']) == "neurio_w2_tcp":
+                meter_manufacturer = "NEURIO"
+            rest = {
+                "componentParentDin": lookup(self.config, ['vin']),
+                "firmwareVersion": None,
+                "lastCommunicationTime": lookup(n, ['timestamp']),
+                "manufacturer": meter_manufacturer,
+                "meterAttributes": {
+                    "meterLocation": []
+                },
+                "serialNumber": sn
+            }
+            neurio[f"NEURIO--{sn}"] = {**cts, **rest}
+        return neurio
+
+    def get_pvac(self):
+        pvac = {}
+        pvac_strings = self.__get_pvac_strings()
+
+        # Loop through each device serial number
+        for i, p in enumerate(lookup(self.status, ['esCan', 'bus', 'PVAC']) or {}):
+            if not p['packageSerialNumber']:
+                continue
+            packagePartNumber = p.get('packagePartNumber', str(i))
+            packageSerialNumber = p.get('packageSerialNumber', str(i))
+            pvac_name = f"PVAC--{packagePartNumber}--{packageSerialNumber}"
+            V_A = p['PVAC_Logging']['PVAC_PVMeasuredVoltage_A']
```

Comment:
p['PVAC_Logging'] can be broken out into a temp variable for clearer readability.

Reply:
Ditto, copied code I'm disinclined to touch

Reply:
Alright, I'll come in with my refactoring later on.